test-data/unit/check-classes.test

```diff
@@ -1223,11 +1223,8 @@ class D:
     def __get__(self, inst: Base, own: Type[Base]) -> str: pass
 [builtins fixtures/bool.pyi]
 [out]
-main:5: error: Revealed type is 'Any'
-main:5: error: No overload variant of "__get__" of "D" matches argument types "None", "Type[A]"
-main:5: note: Possible overload variants:
-main:5: note:     def __get__(self, inst: None, own: Type[Base]) -> D
-main:5: note:     def __get__(self, inst: Base, own: Type[Base]) -> str
+main:5: error: Revealed type is 'd.D'
```

Comment:
I am not sure this is a good idea to have `d.D` instead of `Any`. In case of errors we typically return `Any` to avoid other (potentially bogus) errors.

Reply:
I think in this case, what's happening is that there's actually only one plausible matching overload -- when we do `A.f` and `f` is a descriptor, we always end up calling the overload `def __get__(self, inst: None, own: Type[Base]) -> D`.

So, since only one variant is a plausible match, we just infer that variant's return type. (And regarding your comment below -- this is also why we don't bother showing the possible overload variants here).

I think this behavior is consistent with how we handle bad calls to regular functions. E.g. mypy will infer the return type instead of Any in this program:

    def foo(x: int) -> int: 
        return x

    a = foo("asdf")   # Error
    reveal_type(a)  # Revealed type is 'builtins.int', not Any

Granted, I don't think it's obvious that only one variant will be selected here, so I added another test case for this behavior.

Reply:
OK.